### PR TITLE
rework operator settings

### PR DESCRIPTION
**Pull request by frank-weinberg**
- Make speed score controls always visible
- store remaining option (start/stop buttons) per operator

### CODE REVIEW

File: html/controls/operator.js [hunk 1]
```diff
@@ -147,23 +166,22 @@ function createMetaControlTable() {
...
+	$sb("ScoreBoard.FrontendSettings.Setting(ScoreBoard.Operator.StartStopButtons)").$sbControl(showStartStopButton, { sbcontrol: {
+			button: true
+		}, sbelement: {
+			convert: function(value) {
+				showStartStopButton.filter("input:checkbox")
+				.button("option", "label", (isTrue(value)?"Start/Stop Buttons shown":"Start/Stop Buttons hidden"));
+				$("#TeamTime").find("tr.Control").toggleClass("Show", isTrue(value));
+				var operator = $("#operatorId").text();
+				if (operator) {
+					$sb("ScoreBoard.FrontendSettings.Setting(ScoreBoard.Operator__"+operator+".StartStopButtons)").$sbSet(value);
```
Comment (brian-brazil) [inline html/controls/operator.js]: I'd expect the operator to be in parens here

File: html/controls/operator.js [hunk 2]
```diff
@@ -38,13 +38,30 @@ $sb(function() {
 	$("<button>").text("Logout").click(logout).button().css("float", "right").appendTo("#tabBar");
 });
 
+function setOperatorSettings(op) {
+	var operator;
+	if (op !== "") {
+		operator = "__"+ op + ".";
+	} else {
+		operator = "_Default.";
```
Comment (brian-brazil) [inline html/controls/operator.js]: The default is `default`

**Comment by brian-brazil** [comment]
What happens if two people are logged in at once? I think we might want to make the current operator a persistent setting.

**Comment by frank-weinberg** [comment]
Two operators logged in at once (on different clients) would overwrite each other's settings. (Didn't consider that possibility.)

Making the operator persistent would also have undesirable side effects: If you open a second operator screen and login with a different operator there, the settings and key bindings of the first screen would suddenly change.

So the only proper solutions would be to only allow a single instance of the operator panel to be open or to keep these settings localized in the frontend. The latter is way easier to implement and consistent with current behhaviour.

**Comment by brian-brazil** [comment]
> The latter is way easier to implement and consistent with current behhaviour.

That should work, other than for autostart as it affects the backend.

File: html/controls/operator.js (hunk 1 above)
Comment (frank-weinberg) [reply html/controls/operator.js]: I tried that. It caused the setting name to be nulled somewhere in the XML handling, leading to null pointer exceptions all over the place.

File: html/controls/operator.js (hunk 2 above)
Comment (frank-weinberg) [reply html/controls/operator.js]: This is intentionally different from the setting of the default operator (note also the single vs. double underscore), as it should be a read only value, so new operators will get the default value and not whatever the last person used that didn't log in.

**Comment by frank-weinberg** [comment]
So I guess this will settle our discussion where to put AutoStart in favor of the ruleset, simply out of technical necessity.

File: html/controls/operator.js (hunk 2 above)
Comment (brian-brazil) [reply html/controls/operator.js]: Can you add a comment to that effect?

**Comment by brian-brazil** [comment]
Yeah, if you don't have a persistent single login then it has to go in either FrontendSettings or Ruleset.

File: html/controls/operator.js (hunk 2 above)
Comment (frank-weinberg) [reply html/controls/operator.js]: I had the same thought as I wrote the explanation. It's in there now.

**Comment by frank-weinberg** [comment]
What would still cause interference is when two operator interfaces are open and set to the same operator. But that is probably unavoidable if you want to store the settings in the backend.

File: html/controls/operator.js [hunk 3]
```diff
@@ -122,6 +138,11 @@ function createScoreTimeContent(table) {
 	initialLogin();
 }
 
+function setClockControls(value) {
+	$("#ShowClockControlsButton").checked = value;
```
Comment (brian-brazil) [inline html/controls/operator.js]: This isn't changing the visual display of the button.

Comment (frank-weinberg) [reply html/controls/operator.js]: For me it changes the text color to red (which is the same behaviour as current HEAD).

Comment (brian-brazil) [reply html/controls/operator.js]: Try setting it, and then reloading the page - the start/stop buttons will be there but this won't be red.

Comment (frank-weinberg) [reply html/controls/operator.js]: This should be working now.

Comment (brian-brazil) [reply html/controls/operator.js]: Something's still not right. It's activated now on reload, but it's taking an extra click to toggle it the first time.

Comment (frank-weinberg) [reply html/controls/operator.js]: Finally. (And while the majority of that 1 week delay is attributed to an ugly cold, getting that button to work still took way too much time.)

**Comment by brian-brazil** [comment]
There's something weird going on when you duplicate a tab that's a new behaviour, but it's not due to this change.

**Pull request merged**